crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -209,6 +173,54 @@ impl Trade {
         })
     }
 
+    /// Protocol fee as a cut of surplus, denominated in SURPLUS token
+    fn surplus_fee(&self, factor: f64) -> Result<eth::Asset, Error> {
+        let surplus = self.surplus().ok_or(Error::Surplus(self.sell, self.buy))?;
+        let fee = surplus
+            .amount
+            .apply_factor(factor / (1.0 - factor))
```

Comment:
Why don't we need to differentiate between buy and sell orders here?

Reply:
resolved with added comment